### DISCUSSION TITLE
Can't redirect xontrib output to variable

### DISCUSSION THREAD

**Discussion by Nebucatnetzer** [xonsh/xonsh#4647] [Q&A] [answered]
I try to automatically install xontrib plugins and thought I would check in `xontrib list` before installing a plugin.
E.g. something like this:

```
plugins = $(xontrib list)

if plugin not in plugins:
    xpip install plugin
```

**Comment by jnoortheen** [comment, marked as answer]
It is a bug. That particular alias function doesn't do proper redirection and prints to stdout. 

You can use its Python API call, 

```py
from xonsh.xontribs import xontrib_data

xontrib_data()
```

**Reply by Nebucatnetzer** [reply]
Got it, thank you for your help!
